### PR TITLE
FHG-1982 replace url Params with Redis cache

### PR DESCRIPTION
**Pull request by MartinBelton-gov**
FHG-1982 / FHG-2049 replace url Params with Redis cache when using the connect wizard
Added Unit Tests

### CODE REVIEW

File: src/FamilyHubs.ReferralUi.Ui/FamilyHubs.ReferralUi.Ui.csproj [hunk 1]
```diff
@@ -27,4 +28,15 @@
     <PackageReference Include="Serilog.Sinks.Console" Version="4.1.0" />
     <PackageReference Include="System.IdentityModel.Tokens.Jwt" Version="6.26.0" />
   </ItemGroup>
+
+	<ItemGroup>
+		<AssemblyAttribute Include="System.Runtime.CompilerServices.InternalsVisibleTo">
```
Comment (lazcool) [inline src/FamilyHubs.ReferralUi.Ui/FamilyHubs.ReferralUi.Ui.csproj:33]: It's preferable to only test the public interface (but testing non-public can sometimes be useful).

Comment (MartinBelton-gov) [reply src/FamilyHubs.ReferralUi.Ui/FamilyHubs.ReferralUi.Ui.csproj:33]: I have made the class used by redis Cache internal and obfuscated it to make life a little more difficult for any potential hacker. We might decide to do more at a latter stage

File: src/FamilyHubs.ReferralUi.Ui/Pages/ProfessionalReferral/ContactDetails.cshtml.cs [hunk 2]
```diff
@@ -9,68 +11,61 @@ namespace FamilyHubs.ReferralUi.Ui.Pages.ProfessionalReferral;
...
 
     [BindProperty]
     [Phone(ErrorMessage = "Please enter a valid phone number")]
-    public string? Telephone { get; set; } = default!;
+    public string? Telephone { get; set; }
 
     [BindProperty]
     public bool TelephoneValid { get; set; } = true;
 
+    public string FullName { get; set; } = default!;
```
Comment (lazcool) [inline src/FamilyHubs.ReferralUi.Ui/Pages/ProfessionalReferral/ContactDetails.cshtml.cs:33]: The other strings in the model are nullable.
Does FullName need to be non-nullable and set to default?
Setting a non-nullable to default should only really be used when necessary, such as with types that are used for deserialisation.

Comment (MartinBelton-gov) [reply src/FamilyHubs.ReferralUi.Ui/Pages/ProfessionalReferral/ContactDetails.cshtml.cs]: Replaced FullName with string.Empty, Email, Telephone, Textphone are option, although you must pick one.

**Pull request merged**